envisage/tests/test_extension_point_changed.py

```diff
@@ -188,37 +262,29 @@ def test_assign_empty_list(self):
         self.assertEqual(0, listener.new.index.start)
         self.assertEqual(3, listener.new.index.stop)
 
-    def test_assign_empty_list_no_event(self):
-        """ assign empty list no event """
+    def test_assign_empty_list_with_observe(self):
+        """ assign an empty list to a plugin triggers a list change event."""
 
         a = PluginA()
-        a.on_trait_change(listener, "x_items")
         b = PluginB()
         c = PluginC()
 
+        events = []
+        a.observe(events.append, "x:items")
+
         application = TestApplication(plugins=[a, b, c])
         application.start()
 
         # Assign an empty list to one of the plugin's contributions.
         b.x = []
 
-        # Make sure we pick up the correct contribution via the application.
-        extensions = application.get_extensions("a.x")
-        extensions.sort()
-
-        self.assertEqual(3, len(extensions))
-        self.assertEqual([98, 99, 100], extensions)
-
-        # Make sure we pick up the correct contribution via the plugin.
-        extensions = a.x[:]
-        extensions.sort()
-
-        self.assertEqual(3, len(extensions))
-        self.assertEqual([98, 99, 100], extensions)
-
-        # We shouldn't get a trait event here because we haven't accessed the
-        # extension point yet!
-        self.assertEqual(None, listener.obj)
```

Comment:
This test is removed because it is testing the behaviour noted in the "fixme" comment:
```
# fixme: If the extension point has not been accessed then the
# provider extension registry can't work out what has changed, so it
# won't fire a changed event.
``` 

The "fixme" is now fixed.